### PR TITLE
Support turn off  NodeSourcePlugin with false option

### PR DESCRIPTION
**Pull request by jchip**
**What kind of change does this PR introduce?**

Allow a new value `false` for the `node` option to turn off the `NodeSourcePlugin` completely.  

**Did you add tests for your changes?**

**If relevant, link to documentation update:**

**Summary**

As requested here https://github.com/webpack/webpack/issues/4976, it is desired to have a simple way to turn off `NodeSourcePlugin` completely to avoid any unintentional code to pull in node compatible code for the browser.  Since the plugin supports the `node` option but doesn't have an easy and simple way for turning it off, this PR updates to support that.

**Does this PR introduce a breaking change?**

No

**Other information**

### CODE REVIEW

**Comment by jchip** [comment]
yup, looking... is there some existing tests for `NodeSourcePlugin` I can add to or reference, or should I add a new one under `test/configCases/plugins` or else where?  thanks.

**Comment by jchip** [comment]
@sokra added a simple test.  I can't figure out how to verify if the Plugin's not applied its effects.  Please let me know any pointers you have or any changes you want me to make.  Thanks for reviewing.

File: test/configCases/plugins/node-source-plugin-off/index.js [hunk 1]
```diff
@@ -0,0 +1 @@
+it("should run", function() {});
```
Comment (sokra) [inline test/configCases/plugins/node-source-plugin-off/index.js]: Could you test something like `process.browser === undefined`?

**Comment by sokra** [comment]
Better move the test into `configCases/parsing`

**Pull request merged**